### ISSUE TITLE
field initializers for non-static classes

### ISSUE THREAD

**Issue by mateidavid-work** [OpenHFT/Chronicle-Wire#142]
When extending `AbstractMarshallableCfg`, a list field that is initialized and not de-serialized, is:
- non-null, for a static class - expected
- null, for a non-static class - unexpected, at least to me :)

Perhaps Wire doesn't properly detect the initializer?

```
package local;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.ArrayList;
import java.util.List;
import net.openhft.chronicle.wire.AbstractMarshallableCfg;
import net.openhft.chronicle.wire.Marshallable;
import org.junit.jupiter.api.Test;

class InitializedListInNonStaticClassTest {
    static class StaticData extends AbstractMarshallableCfg {
        int anInt;
        List<String> aList = new ArrayList<>();
    }

    class NonStaticData extends AbstractMarshallableCfg {
        int anInt;
        List<String> aList = new ArrayList<>();
    }

    @Test
    void test() {
        StaticData staticData = Marshallable.fromString(StaticData.class,
            "!StaticData { anInt: 42 }");
        assertThat(staticData).isNotNull();
        assertThat(staticData.anInt).isEqualTo(42);
        assertThat(staticData.aList).isNotNull();   // <== OK, EXPECTED

        NonStaticData nonStaticData = Marshallable.fromString(NonStaticData.class,
            "!NonStaticData { anInt: 42 }");
        assertThat(nonStaticData).isNotNull();
        assertThat(nonStaticData.anInt).isEqualTo(42);
        assertThat(nonStaticData.aList).isNull();   // <== UNEXPECTED
    }
}
```

**Comment by mateidavid-work** [comment]
In `net.openhft:chronicle-wire:2.17.37`.

Maybe it's unusual to read non-static Marshallable classes. I stumbled on this in tests, when trying to do:
```
@Test
void test() {
    class Data extends AbstractMarshallableCfg {...}
    ...
    Data data = Marshallable.fromString(Data.class, "...")
}
```

**Comment by RobAustin** [comment]
As a rule of thumb, I suggest that you make the DTO classes as static classes, the wider problem in chronicle wire may be that we are not reporting this as the underlying issue.

**Comment by RobAustin** [comment]
@peter-lawrey what are your thoughts?

**Comment by peter-lawrey** [comment]
This is expected behaviour. Your non-static class has a reference to the outer class which isn't available so you get a warning like this

```
[main] WARN net.openhft.chronicle.wire.WireMarshaller - Found this$0, in class net.openhft.chronicle.wire.MarshallableTest$NonStaticData which will be ignored!
```

however, it also means the constructor which must be passed a non-null outclass instance can't be called as we don't create a dummy instance of it.

In short, make the class `static`, package local or at the top level. nested classes won't work if you use the outer class instance in any case.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.41](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.41), [BOM-2.20.67](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.67)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
